OpenSSL-Strategy.md

```diff
@@ -0,0 +1,94 @@
+
+In addition, the TLS 1.3 specification is in "Last Call" state, due to end on the 2nd of March. This means that it is likely that we will receive a finalized TLS 1.3 specification during March, which would make a May OpenSSL 1.1.1 release feasable.
+
+TLS 1.3 specification finalization and code readiness are the two reasons that OpenSSL 1.1.1 release in early May is not a certainty. However, current signs point to that probability.
+
+## FIPS
+
+The [Federal Information Processing Standard](https://en.wikipedia.org/wiki/FIPS_140-2) is managed by the NIST. Publication 140-2 is a US government standard used to approve cryptographic software.
+
+Various organizations rely on the FIPS standard and mandate that software that they use conform appropriately. OpenSSL itself is not FIPS "validated", instead a separate "software component" called the "OpenSSL FIPS Object Module" has been created for the purpose of validation. It is a drop-in replacement for OpenSSL such that software using the OpenSSL API can be adapted to use it. In order to retain FIPS compliance, the OpenSSL FIPS Object Module source code must be obtained via a secure path as dictated by a security policy and the source must be compiled according to specific steps without variation.
+
+Node.js compiled against the OpenSSL FIPS Object Module does not make Node.js itself FIPS validated. Instructions for compiling Node.js against the Object Module along with links to further documentation are detailed in [BUILDING.md](https://github.com/nodejs/node/blob/master/BUILDING.md#building-nodejs-with-fips-compliant-openssl).
+
+The OpenSSL FIPS Object Module is compatible with OpenSSL 1.0.2 and Node.js has been able to build with this module since 2015, prior to Node.js 4. It requires some modification of the Node.js internals (see `git grep FIPS -- lib/ src/`) for this to work properly.
+
+Development and validation of a FIPS software component is time consuming and expensive. The OpenSSL team have yet to commit to a timeframe for development of the next generation of its FIPS Object Module, however they have stated that it is their next priority ["after 1.1.1"](https://www.openssl.org/policies/roadmap.html). Therefore, any user requiring FIPS validated OpenSSL must use OpenSSL 1.0.2 until a new module is made available that supports a newer version of OpenSSL.
+
+## Node.js 10
+
+OpenSSL LTS support timing, the lack of OpenSSL LTS planning and the lack of a clear timeframe for a new FIPS module complicates Node.js 10.
+
+As of the time of writing, the strategy for OpenSSL with Node.js 10 is:
+
+ * OpenSSL 1.1.0 to initially be the assumed default compile target.
+ * Bundle a copy of OpenSSL 1.1.0 in the source tree, along with any floating patches still required for improved Windows support and test-runner speed-ups.
```

Comment:
I think we should add the following a new build requirement especially for Windows.

* A new build requirement of NASM is needed on Windows as in 
https://github.com/openssl/openssl/blob/89314d9a22ad0b0ce87e188c486532926aff01ea/NOTES.WIN#L21-L24 .
Microsoft assembler, `masm`, is no longer supported in building Node.js with OpenSSL-1.1.x.

Reply:
A newer gas or llvm version would be required for use of AVX512 features but it has slightly different version requirements between 1.1.0 and 1.1.1.
 I think it is one the issues to be resolved before include OpenSSL-1.1.x build binding with Node

Reply:
resolved by updating BUILDING.md, see
- https://github.com/nodejs/node/pull/19794/commits/3f131489365dc83ba6adcdcc2a98fbdb6f63b4b5
- https://github.com/nodejs/node/pull/19794/commits/f584f354770ab44c79c1294f81a915f255d677e2
- https://github.com/nodejs/node/pull/25381/commits/5703c7f9a657b361e8ea5f8b6eb1174fe5598a64